src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.ProductDao;
+import core.basesyntax.dao.ProductDaoImpl;
+import core.basesyntax.operations.BalanceOperation;
+import core.basesyntax.operations.Operation;
+import core.basesyntax.operations.Operations;
+import core.basesyntax.operations.PurchaseOperation;
+import core.basesyntax.operations.ReturnOperation;
+import core.basesyntax.operations.SupplyOperation;
+import core.basesyntax.service.CsvReaderServiceImpl;
+import core.basesyntax.service.CsvWriterServiceImpl;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.service.FruitShopServiceImpl;
+import core.basesyntax.service.ReaderService;
+import core.basesyntax.service.WarehouseService;
+import core.basesyntax.service.WarehouseServiceImpl;
+import core.basesyntax.service.WriterService;
+import core.basesyntax.strategy.OperationsStrategy;
+import core.basesyntax.strategy.OperationsStrategyImpl;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Main {
+    private static final String FILE_PATH_FROM = "src/main/resources/file.csv";
+    private static final String FILE_PATH_TO = "src/main/resources/report.csv";
+
+    public static void main(String[] args) {
+        Map<Operations, Operation> operationMap = new HashMap<>();
+        operationMap.put(Operations.B, new BalanceOperation());
+        operationMap.put(Operations.S, new SupplyOperation());
+        operationMap.put(Operations.P, new PurchaseOperation());
+        operationMap.put(Operations.R, new ReturnOperation());
+
+        OperationsStrategy operationsStrategy = new OperationsStrategyImpl(operationMap);
+        ProductDao productDao = new ProductDaoImpl();
+        WarehouseService warehouseService = new WarehouseServiceImpl(operationsStrategy,
+                productDao);
+
+        ReaderService readerService = new CsvReaderServiceImpl();
+        WriterService writerService = new CsvWriterServiceImpl();
+
+        FruitShopService fruitShopService = new FruitShopServiceImpl(warehouseService,
+                readerService, writerService);
+        fruitShopService.createReport(FILE_PATH_FROM, FILE_PATH_TO);
```

Comment:
let’s not make your `fruitShopService` dependent on reader and writer - we should make our services independent

let’s call your services one by one and use data returned from one service - to pass into next service
use your reader to read lines - return `List<String>`
then pass `List<String>` into fruitShopService for saving data to storage
call `fruitShopService.getReport()` to get report from storage as String
pass report and path to output file into writer service

your services will be independent

in fact, you won't need both `warehouseService` and `fruitShopService` - we can leave only one of them

Reply:
done